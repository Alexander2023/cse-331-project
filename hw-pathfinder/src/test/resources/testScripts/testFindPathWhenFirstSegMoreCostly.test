# Tests that path finding applies Dijkstra's algorithm properly
# by finding the overall shortest path, even when the first segment
# is more costly

CreateGraph g

AddNode g p1
AddNode g p2
AddNode g p3
AddNode g p4

AddEdge g p1 p2 1.0
AddEdge g p2 p4 3.0
AddEdge g p1 p3 2.0
AddEdge g p3 p4 1.0

FindPath g p1 p4