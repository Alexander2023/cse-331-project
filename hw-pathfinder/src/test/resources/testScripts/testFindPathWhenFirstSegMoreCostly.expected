# Tests that path finding applies Dijkstra's algorithm properly
# by finding the overall shortest path, even when the first segment
# is more costly

created graph g

added node p1 to g
added node p2 to g
added node p3 to g
added node p4 to g

added edge 1.000 from p1 to p2 in g
added edge 3.000 from p2 to p4 in g
added edge 2.000 from p1 to p3 in g
added edge 1.000 from p3 to p4 in g

path from p1 to p4:
p1 to p3 with weight 2.000
p3 to p4 with weight 1.000
total cost: 3.000